src/interactive/repl.ts

```diff
@@ -376,6 +376,97 @@ function clearProgress() {
     }
 }
 
+function display(params: { kind: string, data: any }) {
+    if (params.kind === 'application/vnd.julia-vscode.diagnostics') {
+        displayDiagnostics(params)
+    } else {
+        plots.displayPlot(params)
+    }
+}
+
+interface diagnosticData {
+    msg: string,
+    path: string,
+    line?: number,
+    range?: number[][],
```

Comment:
That seems strange to me, to have both `line` and `range`. I think we should just mimic the diagnostic API in VS Code here, so have one field `range` that has fields `start` and `end`, and these each have a field `line` and `character`.

Reply:
Either is possible. `range` takes precedence. Both `[[], []]` and the `Range`/`Position` formats are used in VSCode. This just mimics the VSCode API we're using later on. Generally I'd like to have as few types as possible here.